src/groupByKey.ts

```diff
@@ -2,6 +2,18 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function
+groupByKey<T, K extends keyof T>(items: T[], key: K): GroupsMap<T> {
+  const result: GroupsMap<T> = items.reduce((acc, ch) => {
+    const keyValue = ch[key] as unknown as string;
```

Comment:
Avoid using 'as unknown as string'. It's better to use type guards or type assertions properly to avoid any possible type errors.